builtin/docker/platform.go

```diff
@@ -141,11 +141,22 @@ func (p *Platform) Deploy(
 	s.Done()
 
 	s = sg.Add("Creating new container")
+	defer s.Abort()
```

Comment:
is that true even though this is a different "Step" at this point than the step created previously?

Reply:
Yep, I believe so. You'll notice else where on Waypoint we only defer once but add multiple steps (you can see multiple examples of this in the `serverinstall` package.). I might be wrong on this but I know @mitchellh or @evanphx would know better!

Reply:
Hey @briancain any word from @mitchellh or @evanphx  internally about this?

For reference, here's what the behavior looks like in a test with and without the additional Abort()
https://gist.github.com/piotrromanowski/ca68fd1fdb75170c64a40f377fb905a8

And when running locally, here's a screenshot of what it looks like if you return early (with an error) with and without the `defer Abort()`. Notice that you see a clear error in the first run, but the second run just hangs for me indefinitely. 

https://imgur.com/22JEMi7

Reply:
@piotrromanowski I'll give your branch a shot today and see if I can get a better answer for you :) Thanks for the extra info here! I'll let you know what I find. Otherwise I'd say this PR is good to go 👍🏻

Reply:
Just gave it a shot, you're right here! Looks like it was an existing bug too. I'll dismiss this then.